homeassistant/components/airly/sensor.py

```diff
@@ -60,26 +72,49 @@ def __init__(
     ) -> None:
         """Initialize."""
         super().__init__(coordinator)
-        description = SENSOR_TYPES[kind]
-        self._attr_device_class = description[ATTR_DEVICE_CLASS]
-        self._attr_extra_state_attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
-        self._attr_icon = description[ATTR_ICON]
+        self._description = description = SENSOR_TYPES[kind]
+        self._attr_device_class = description.get(ATTR_DEVICE_CLASS)
+        self._attr_icon = description.get(ATTR_ICON)
         self._attr_name = f"{name} {description[ATTR_LABEL]}"
-        self._attr_state_class = description[ATTR_STATE_CLASS]
+        self._attr_state_class = description.get(ATTR_STATE_CLASS)
         self._attr_unique_id = (
             f"{coordinator.latitude}-{coordinator.longitude}-{kind.lower()}"
         )
-        self._attr_unit_of_measurement = description[ATTR_UNIT]
+        self._attr_unit_of_measurement = description.get(ATTR_UNIT)
+        self._attrs: dict[str, Any] = {ATTR_ATTRIBUTION: ATTRIBUTION}
         self.kind = kind
         self._state = None
 
     @property
     def state(self) -> StateType:
         """Return the state."""
         self._state = self.coordinator.data[self.kind]
```

Comment:
`self._state` instance attribute isn't needed now. We can just use a local variable.

Reply:
```suggestion
        state = self.coordinator.data[self.kind]
```